### ISSUE TITLE
Improve UX of modal alert/confirmation dialog windows

### ISSUE THREAD

**Issue by andrewhayward** [WordPress/gutenberg#59987]
Relating to both #58390 and #58690, we need to give modal alert/confirmation dialogs some UX attention:

- [x] **Remove the `×` cancel button**
  It's super confusing when the only options are `×`, <kbd>Cancel</kbd> and <kbd>{Action}</kbd>.
- [x] **Ensure destructive actions use an `isDestructive` button**
  It is important that we denote when an action has a negative consequence
- [x] **Ensure keyboard focus is visible**
  If the user has opened a dialog with the keyboard, it should be clear where the focus now is
- [x] **Focus constructive buttons by default**
  If a dialog window serves a specific purpose, users should have minimal friction to complete it
- [x] ***Do not* focus destructive buttons by default**
  If a confirmation button has a negative action, we should make sure it does not have initial focus

| Before | After |
|--------|-------|
| ![A dialog window, with an "x" in the top right corner, a "Cancel" button, and a blue "Delete" button](https://github.com/WordPress/gutenberg/assets/159848/0b397c5c-cea0-4eac-affe-5eed953d9bd6) | ![A dialog window, with a "Cancel" button, and a red "Delete" button](https://github.com/WordPress/gutenberg/assets/159848/faa600b3-342a-4b1f-a89f-1c56466e07a4) |

**Comment by jasmussen** [comment]
> Ensure destructive actions use an isDestructive button
> It is important that we denote when an action has a negative consequence

IMO this is redundant in a modal such as this, where the entire reason for existing is to indicate the destructive action. To that end, the color change _here_ simply serves to ensure there are two colors of buttons in the modal, which _could_ add confusion: "why is this red?" Furthermore as just a color change it's not that useful to people with color blindness, whereas the modal itself should be context enough.

Not a strong opinion, happy to defer.

**Comment by mirka** [comment]
The points in the issue description are all implemented now in [`AlertDialog`](https://wordpress.github.io/gutenberg/?path=/docs/design-system-components-alertdialog--docs).
